### PR TITLE
[NO-TICKET] Fix known typing bugs

### PR DESCRIPTION
**Pull request by pwolfert**
## Summary
Integration tests for three downstream products were failing due to these TypeScript errors.

This will be cherry-picked onto the current beta release, and we'll release a new beta.

### Fixed
- Fixed optional prop in Dialog being marked as required in TypeScript definitions
- Fixed optional props in HelpDrawer being marked as required in TypeScript definitions
- Fixed TypeScript example project missing required props for Pagination

## How to test
1. Build core package with `yarn build`
2. In the `examples/create-react-app-typescript` example project, set the core package version to `file:../../packages/design-system`
3. Clear the example's `node_modules` if it's old
4. In `examples/create-react-app-typescript`, `yarn install`
5. In `examples/create-react-app-typescript`, `yarn start`. The example app should work

### CODE REVIEW

File: examples/create-react-app-typescript/package.json [hunk 1]
```diff
@@ -16,9 +16,6 @@
     "start": "react-scripts start",
     "build": "react-scripts build"
   },
-  "eslintConfig": {
-    "extends": "react-app"
-  },
```
Comment (pwolfert) [inline examples/create-react-app-typescript/package.json:21]: Right now there are errors committing due to eslint, so disable eslinting for the example until we have a chance to audit our lint rules.

Comment (pwolfert) [reply examples/create-react-app-typescript/package.json:21]: @scheul93, you were getting this same error earlier this week. Do you agree with the temporary fix to just remove the linting for the example?

**Review by forrestbaer — APPROVED** [review]
eyuuuup!

File: examples/create-react-app-typescript/package.json (hunk 1 above)
Comment (scheul93) [reply examples/create-react-app-typescript/package.json:21]: Seems fine to me. Especially because these examples are more for internal testing purposes

**Review by scheul93 — APPROVED** [review]
LGTM

**Pull request merged**